sync/mock.go

```diff
@@ -0,0 +1,77 @@
+package sync
+
+import (
+	"github.com/weaveworks/flux"
+	"github.com/weaveworks/flux/policy"
+)
+
+type rsc struct {
+	bytes []byte
+	Kind  string
+	Meta  struct {
+		Namespace string
+		Name      string
+	}
+}
+
+type rscIgnorePolicy struct {
```

Comment:
Can you get the same effect by embedding `rsc` in this struct and overriding only the method you need, rather than recapitulating its fields?

Reply:
I cannot confirm at the moment as my local env is broken. Will hopefully sort out tomorrow.

Reply:
My local env is broken, so will do it as soon as it's fixed.